File: EduMS.Backend/src/EduMS.Application/M6_StatisticsReports/DTOs/ComparativeReports/ComparativeReportDto.cs
using System;

namespace EduMS.Application.M6_StatisticsReports.DTOs.ComparativeReports;

public class ComparativeReportDto
{
    public long Id { get; set; }
    public long SchoolId { get; set; }
    public string ReportNumber { get; set; } = string.Empty;
    public string ComparisonTitle { get; set; } = string.Empty;
    public string FirstPeriodLabel { get; set; } = string.Empty;
    public string FirstPeriodStart { get; set; } = string.Empty;
    public string FirstPeriodEnd { get; set; } = string.Empty;
    public string SecondPeriodLabel { get; set; } = string.Empty;
    public string SecondPeriodStart { get; set; } = string.Empty;
    public string SecondPeriodEnd { get; set; } = string.Empty;
    public string ComparisonType { get; set; } = string.Empty;
    public string? KpiComparedJson { get; set; }
    public string? ComparisonDataJson { get; set; }
    public string? AutoInsights { get; set; }
    public string? Summary { get; set; }
    public DateTime GenerationDate { get; set; } = DateTime.UtcNow;
    public long? GeneratedByUserId { get; set; }
    public string? FileFormat { get; set; }
    public string? FilePath { get; set; }
    public int ViewCount { get; set; }
    public DateTime? LastViewedAt { get; set; }
    public int ReportStatus { get; set; } = 1;
    public string? Notes { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public long CreatedByUserId { get; set; }
    public DateTimeOffset? ModifiedAt { get; set; }
    public long? ModifiedByUserId { get; set; }
    public bool IsDeleted { get; set; }
    public DateTimeOffset? DeletedAt { get; set; }
    public long? DeletedByUserId { get; set; }
    public Guid VersionToken { get; set; }
    public DateTimeOffset? LastSyncedAt { get; set; }
    public string SyncStatus { get; set; } = string.Empty;
}

File: EduMS-Frontend/edums-frontend-DDD-System/src/app/modules/m6-statistics-reports/interfaces/comparative-report.dto.ts
import { BaseAuditFields, ComparisonType, FileFormat, ReportStatus } from './base.types';

export interface ComparativeReport extends BaseAuditFields {
  schoolId: number;
  reportNumber: string;
  comparisonTitle: string;
  firstPeriodLabel: string;
  firstPeriodStart: string;
  firstPeriodEnd: string;
  secondPeriodLabel: string;
  secondPeriodStart: string;
  secondPeriodEnd: string;
  comparisonType: ComparisonType;
  kpiComparedJson: string | null;
  comparisonDataJson: string | null;
  autoInsights: string | null;
  summary: string | null;
  generationDate: string;
  generatedByUserId: number | null;
  fileFormat: FileFormat | null;
  filePath: string | null;
  viewCount: number;
  lastViewedAt: string | null;
  reportStatus: ReportStatus;
  notes: string | null;
}

export type CreateComparativeReport = Omit<ComparativeReport, 'id' | 'createdAt' | 'modifiedAt' | 'viewCount' | 'lastViewedAt'>;

export type UpdateComparativeReport = Pick<ComparativeReport, 'id'> & Partial<Omit<ComparativeReport, 'id' | 'createdAt' | 'modifiedAt'>>;
